### PR TITLE
Add support for syntax highlighting

### PR DESCRIPTION
**Pull request by jnordberg**
Only enabled for gfm language tagged codeblocks, e.g. **```javascript**

Using https://highlightjs.org

### CODE REVIEW

**Comment by goldibex** [comment]
@roadscape please review per @sneak above

**Comment by trashhalo** [comment]
Is the pr dead on the vine? Would be lovely to get syntax highlighting merged as I'd like to do programming tutorials on steem.

**Comment by bnchdrff** [comment]
highlightjs seems to be pretty solid, and has no huge runtime deps.. any other comments?

i, also, would love to have this feature.

**Comment by jnordberg** [comment]
Ping @sneak @roadscape

**Comment by roadscape** [comment]
Anything in the rendering path requires in-depth security review. A similar request we had was MathJax integration, which was deemed impractical to implement safely (https://github.com/steemit/condenser/issues/7). A cursory check of highlight.js suggests it's safe to use but we need a thorough review.

**Comment by sneak** [comment]
I’m for this, if someone qualified reviews the version of highlight.js we are using and @goldibex signs off on the bundle size increase.

File: src/app/components/cards/MarkdownViewer.jsx [hunk 1]
```diff
@@ -13,7 +14,14 @@ const remarkable = new Remarkable({
     breaks: true,
     linkify: false, // linkify is done locally
     typographer: false, // https://github.com/jonschlinkert/remarkable/issues/142#issuecomment-221546793
-    quotes: '“”‘’'
+    quotes: '“”‘’',
+    highlight: (src, lang) => {
+        if (lang && hljs.getLanguage(lang)) {
+            try {
+                return hljs.highlight(lang, src).value
+            } catch (e) {}
```
Comment (goldibex) [inline src/app/components/cards/MarkdownViewer.jsx]: error needs caught and handled properly

**Comment by relativityboy** [comment]
With the bundle size increase of hjs and changes coming to condenser I'm closing this and opening issue #2095 for syntax highlighting in its place.

**Comment by bnchdrff** [comment]
what is the bundle size increased by?

File: src/app/components/cards/MarkdownViewer.jsx (hunk 1 above)
Comment (jnordberg) [reply src/app/components/cards/MarkdownViewer.jsx]: What do you consider properly in this case? We fall back to not syntax highlighting if the method throws an error.

**Comment by jnordberg** [comment]
I would like to know that as well... I think it would not be that a large of an increase, and if it is to big highlight.js can be built with fewer supported languages. @relativityboy I'm reopening this ;)

**Comment by bnchdrff** [comment]
closing for now; we can reopen when we're ready to work on this

**Comment by jnordberg** [comment]
😢

**Comment by bnchdrff** [comment]
the day will come @jnordberg !!!

**Pull request closed without merging**